src/renderer.rs

```diff
@@ -320,4 +325,71 @@ impl Renderer {
             line(color::BLACK.into(), 1., [start[0], start[1], end[0], end[1]], c.transform, g);
         }
     }
+
+    /// Save the drawings to SVG
+    fn save_svg<P: AsRef<path::Path>>(&mut self, path: P) {
+        let drawing_state = self.app.drawing();
+        let mut document = svg::Document::new()
+            .set("viewBox", (0, 0, drawing_state.width, drawing_state.height));
+
+        // set background color - https://stackoverflow.com/a/11293812/9276882
+        let bg_color = drawing_state.background;
+        let background = SvgRect::new()
+            .set("width", "100%")
+            .set("height", "100%")
+            .set("fill", format!("rgba({},{},{},{})", bg_color.red, bg_color.green, bg_color.blue, bg_color.alpha));
+
+        document = document.add(background);
+
+        let center = Point {x: drawing_state.width as f64 * 0.5, y: drawing_state.height as f64 * 0.5};
```

Comment:
The center is actually defined as part of the `drawing_state` variable. I suggest using that property directly instead of computing this.

https://github.com/sunjay/turtle/blob/ccffb27cb003f1b278463511e381ef0dc3c9c8a1/src/state.rs#L61-L70